awsutil/rotate.go

```diff
@@ -114,6 +114,20 @@ func (c *CredentialsConfig) CreateAccessKey(opt ...Option) (*iam.CreateAccessKey
 		return nil, fmt.Errorf("nil AccessKeyId or SecretAccessKey returned from aws.CreateAccessKey")
 	}
 
+	// Check the credentials to make sure they are usable. We only do
+	// this if withTimeout is non-zero to ensue that we don't
+	// immediately fail due to eventual consistency.
+	if opts.withTimeout != 0 {
+		newC := &CredentialsConfig{
+			AccessKey: *createAccessKeyRes.AccessKey.AccessKeyId,
+			SecretKey: *createAccessKeyRes.AccessKey.SecretAccessKey,
+		}
+
+		if _, err := newC.GetCallerIdentity(WithTimeout(opts.withTimeout)); err != nil {
```

Comment:
So thinking about this, the one thing I'm worried about is the exponential backoff. The reason is that this often takes 3-5 seconds but you could easily end up sleeping 4-8 seconds more than you need to. I wonder if we should just fix the retry period at something shorter (maybe 1 or 2 seconds).

Reply:
Every second for a few seconds does not sound like a big deal! I'll make the change.